src/core/network/client.ts

```diff
@@ -346,9 +346,17 @@ class ApiClient {
     }
 
     debug('HTTP call', method, payload)
-    const buildConfig = includesMedia(payload)
-      ? buildFormDataConfig({ method, ...payload }, options.agent)
-      : buildJSONConfig(payload)
+
+    const hasMedia = includesMedia(payload)
+    let buildConfig: Promise<RequestInit>
+    const configPayload: { [key: string]: unknown } = payload
+    if (hasMedia) {
+      configPayload[method] = method
+      buildConfig = buildFormDataConfig(configPayload, options.agent)
+    } else {
+      buildConfig = buildJSONConfig(payload)
+    }
```

Comment:
...what does it improve?

Reply:
It makes the code compile. I have no idea why it suddenly didn't anymore. Do let me know what the issue was in case you decide to investigate this.

Reply:
`Type '{ method: M; } & Opts<M>' is not assignable to type '{ [key: string]: unknown; }'.ts(2322)` was the error, perhaps related to https://devblogs.microsoft.com/typescript/announcing-typescript-3-5/#k-string-unknown-is-no-longer-a-wildcard-assignment-target